gprofiler/utils.py

```diff
@@ -395,20 +396,28 @@ def _take_lock() -> Optional[socket.socket]:
                 raise
 
             # already taken :/
-            return None
+            return False, None
         else:
             # don't let child programs we execute inherit it.
             fcntl.fcntl(s, fcntl.F_SETFD, fcntl.fcntl(s, fcntl.F_GETFD) | fcntl.FD_CLOEXEC)
 
-            return s
+            return True, s
 
-    s = run_in_ns(["net"], _take_lock)
-
-    global gprofiler_mutex
-    # hold the reference so lock remains taken
-    gprofiler_mutex = s
+    res = run_in_ns(["net"], _take_lock)
+    if res is None:
+        # exception in run_in_ns
+        print("Could not acquire gProfiler's lock due to an error. Are you running gProfiler in privileged mode?")
+        return False
+    elif res[0]:
+        assert res[1] is not None
 
-    return s is not None
+        global gprofiler_mutex
+        # hold the reference so lock remains taken
+        gprofiler_mutex = res[1]
+        return True
+    else:
+        print("Could not acquire gProfiler's lock. Is it already running?", file=sys.stderr)
```

Comment:
here too

Reply:
We can be more elaborate here and write which PID holds the lock.

Reply:
Already `sys.stderr`

Reply:
> We can be more elaborate here and write which PID holds the lock.

Hmm, it's kinda complicated to run that. I'll drop in better comment